system/ee/legacy/libraries/Template.php

```diff
@@ -3504,7 +3504,13 @@ public function parse_variables_row($tagdata, $variables, $solo = true)
         }
 
         // same with modified, sometimes devs run this method themselves instead of a full parse_variables()
-        if ($this->modified_vars === false) {
+        /*
+        Reason for if condtion change:
+        mod.comment switched to using this function instead of parse_variables function and now no Variable Modifiers work in comments
+        i.e. {comment:limit characters="20"} will not parse correctly. The getModifiedVariables() is never run here and is run without any checks
+        in parse_variables().  Keep the $this->modified_vars === false condition because of comment above.
+        */
+        if (empty($this->modified_vars) || $this->modified_vars === false) {
```

Comment:
I think `$this->modified_vars === false` will always return true if the first condition is met (`empty($this->modified_vars)`). Or do I miss something here.

Also I don't think we need that comment in the code. It is descriving the change and not code behaviour, and probably would work better if converted to commit message

Reply:
I think this is fine. We could get rid of the `|| $this->modified_vars === false` because I think it's the opposite, that `empty($this->modified_vars)` would be true if `$this->modified_vars === false`. Either way, I don't see an issue leaving this here.